### ISSUE TITLE
rule for special/internal id.

### ISSUE THREAD

**Issue by youngsofun** [databendlabs/databend#13778]
**Summary**

case 1:  column name for column pos
case 2:  special catalog and database name, so TableEntry for stage table of stage `s1` can be quickly identified by  `$STAGE.$STAGE.s1`
case 3: special column name like `METADATA$FILENAME` for metadata of staged files. https://docs.snowflake.com/en/user-guide/querying-metadata (no problem for snowflake since it only support query by pos https://docs.snowflake.com/en/user-guide/querying-stage#query-syntax-and-parameters)

since:
```
    #[regex(r#"[_a-zA-Z][_$a-zA-Z0-9]*"#)]
    Ident,
```

internal id can start with `$`.
or maybe some prefix like `DATABEND_`
cc @andylokandy @sundy-li 

but this still can not avoid confilcting with user column names,  e.g.  
1. quoted id 
2. column of parquet/json files

**Comment by youngsofun** [comment]
> case 2: special catalog and database name, so TableEntry for stage table of stage s1 can be quickly identified by $STAGE.$STAGE.s1

this make it easy to checking privilage of `select from stage`. cc @TCeason

**Comment by TCeason** [comment]
> > case 2: special catalog and database name, so TableEntry for stage table of stage s1 can be quickly identified by $STAGE.$STAGE.s1
> 
> this make it easy to checking privilage of `select from stage`. cc @TCeason

Yes. I agree with this. It makes sense.

If not, I need to do this in check:

![image](https://github.com/datafuselabs/databend/assets/33082201/bbc4c6eb-88de-4949-b9f0-bb3c24391ecb)

**Comment by andylokandy** [comment]
> or maybe some prefix like `DATABEND_`

Indeed, I agree. A more intricate prefix such as `__DATABEND_` or perhaps even `__DATABEND_INTERNAL_` could help prevent conflicts with users' namespaces in practical applications.

**Comment by TCeason** [comment]
Already support
